src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,23 +1,34 @@
-import React from "react";
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
+import { getTokenOwner } from "./../../../services/token";
+import { useEscrowContractUsers } from "../../../hooks";
+import { WrappedDocument } from "@govtechsg/open-attestation";
 
-export const TitleTransferPanelContent = () => {
-  const { holderAddress, beneficiaryAddress } = useSelector((state: any) => ({
-    holderAddress: state.token.holderAddress,
-    beneficiaryAddress: state.token.beneficiaryAddress
-  }));
+interface TitleTransferPanelContentProps {
+  document: WrappedDocument;
+}
+
+export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
+  const [tokenOwnerAddress, setTokenOwnerAddress] = useState("");
+  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwnerAddress });
+  const fetchTokenOwner = async () => {
+    try {
+      const address = await getTokenOwner({ document });
+      setTokenOwnerAddress(address);
+    } catch (e) {
+      console.log(e.message);
+    }
+  };
+  fetchTokenOwner();
 
   return (
     <>
       <div className="row">
-        {(holderAddress || beneficiaryAddress) && (
-          <div className="col-12 col-lg">
-            <TitleView role="Beneficiary" address={beneficiaryAddress} />
-            <TitleView role="Holder" address={holderAddress} />
-          </div>
-        )}
+        <div className="col-12 col-lg">
+          <TitleView id="beneficiary" role="Beneficiary" address={beneficiaryAddress} />
```

Comment:
Why need an id here that is passed a few levels deep? Possible to use other selector using numbered ones? 

Seemed like there's the `nth` selector
![image](https://user-images.githubusercontent.com/7406870/77401663-48d54c80-6de8-11ea-9f51-a4678981fd4d.png)

Reply:
okay 👌